sdk/spring/spring-cloud-azure-feature-management/src/main/java/com/azure/spring/cloud/feature/management/filters/TimeWindowFilter.java

```diff
@@ -30,38 +31,29 @@ public final class TimeWindowFilter implements FeatureFilter {
      */
     @Override
     public boolean evaluate(FeatureFilterEvaluationContext context) {
-        String start = (String) context.getParameters().get(TIME_WINDOW_FILTER_SETTING_START);
-        String end = (String) context.getParameters().get(TIME_WINDOW_FILTER_SETTING_END);
-
-        ZonedDateTime now = ZonedDateTime.now();
+        final ObjectMapper objectMapper = JsonMapper.builder()
+            .configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true).build();
+        final TimeWindowFilterSettings settings = objectMapper.convertValue(context.getParameters(), TimeWindowFilterSettings.class);
+        final ZonedDateTime now = ZonedDateTime.now();
 
-        if (!StringUtils.hasText(start) && !StringUtils.hasText(end)) {
+        if (settings.getStart() == null && settings.getEnd() == null) {
```

Comment:
If want to use recurrence parameters, then "start" and "end" paramter are required parameter.  The recurrence time range is defined by the "start" and "end" parameters. 

For example, if I want to define an event occurs at 2:00AM -- 3:00Am on every Monday from 2023 13th Dec. The config json should be:
```
{
    "start": "2023-12-18T02:00:00+08:00",
    "end": "2023-12-18T03:00:00+08:00" ,
    "recurrence": {
        "pattern": {
            "type": "Weekly",
            "interval": 1,
            "daysOfWeek": [
                "Monday"
            ]
        },
        "range": {
            "type": "NoEnd"
        }
    }
}
```
The "start" and "end" parameter need to specify the time window(2:00 AM -- 3:00 AM) and also match the pattern(2023 18th Dec is Monday)

Reply:
I think you misunderstood. You've modified the time window filter so the old. type doesn't work. I should still be able to do:

```json
{
    "start": "2023-12-31T00:00:00+08:00"
}
```

This time window filter turns on at midnight on the 31st, and never goes off. And the same with this:

```json
{
    "end": "2023-12-31T00:00:00+08:00"
}
```

Where this one start on then turns off when it reaches that date.

Reply:
I'm a little confused about that. We still support only set the start parameter. It will be return false for the following settings.
```
{ "start": "2023-12-31T00:00:00+08:00" }
```

And also support only set end parameter. It will return true for the following settings.
```
{ "end": "2023-12-31T00:00:00+08:00" }
```